### PR TITLE
Various docs fixes: Copy (from|to), configuration, and psql 

### PR DESCRIPTION
**Pull request by celaus**

### CODE REVIEW

File: blackbox/docs/protocols/postgres.txt [hunk 1]
```diff
@@ -5,17 +5,15 @@ Postgres Wire Protocol
 
 Crate contains experimental support for the `PostgreSQL wire protocol v3`_.
 
-If a node is started with postgres support enabled it will by default bind to
-``0.0.0.0:5432``. See :doc:`Configuration <../configuration>` for options to
-change that.
+If a node is started with postgres support enabled it will bind to
+``0.0.0.0:5432`` by default. For Crate to use a custom port, the
+setting the corresponding :ref:`conf_ports`
```
Comment (matriv) [inline blackbox/docs/protocols/postgres.txt]: something missing... maybe: the setting **in** the corresponding?

File: blackbox/docs/protocols/postgres.txt [hunk 2]
```diff
@@ -5,17 +5,15 @@ Postgres Wire Protocol
...
-The SQL dialect that Crate supports is different to the SQL dialect that
-PostgreSQL supports. So even though the wire protocol is supported it doesn't
-imply that any postgres client can be used as those clients may use SQL
-statements which aren't supported.
-
-One notable difference is that Crate doesn't support transactions.
-Therefore clients should generally enable ``autocommit``.
+However, even though connecting PostgreSQL tools and client libraries is
+supported, the actual SQL statements have to be supported by Crate's SQL
+dialect. A notable difference there is that Crate doesn't support transactions,
```
Comment (matriv) [inline blackbox/docs/protocols/postgres.txt]: either: "There is a notable difference..." or: "A notable difference is that ..."

File: blackbox/docs/protocols/postgres.txt [hunk 3]
```diff
@@ -5,17 +5,15 @@ Postgres Wire Protocol
...
+which is why clients should generally enable ``autocommit``.
```
Comment (matriv) [inline blackbox/docs/protocols/postgres.txt]: I prefer "therefore" instead of "which is why"

File: blackbox/docs/protocols/postgres.txt [hunk 4]
```diff
@@ -34,21 +32,18 @@ Therefore clients should generally enable ``autocommit``.
 Server Compatibility & Implementation Status
 ============================================
 
-Crate emulates server version ``9.5``
+Crate emulates a PostgreSQL server version ``9.5``
```
Comment (matriv) [inline blackbox/docs/protocols/postgres.txt]: Would lose the **a**

File: blackbox/docs/protocols/postgres.txt [hunk 5]
```diff
@@ -34,21 +32,18 @@ Therefore clients should generally enable ``autocommit``.
...
-
-The ``database`` name a client sends will be used as default schema as Crate
-doesn't have database support.
+will always respond with ``AuthenticationOK`` and any SSLRequests are answered
+with ``N`` to indicate that it does not support SSL.
 
-The server will inform the Client that ``UTF8`` encoding should be used. Any
-other encoding is not supported.
+Since Crate uses schemas instead of databases, the ``database`` parameter sets
+the default schema name for future queries. Usually this schema is ``doc``.
```
Comment (matriv) [inline blackbox/docs/protocols/postgres.txt]: I wouldn't say usually but: Default schema if no database parameter is specified

File: blackbox/docs/protocols/postgres.txt [hunk 6]
```diff
@@ -67,27 +62,25 @@ The `Extended Query`_ protocol is implemented with the following limitations:
...
-- In order to optimize the execution of bulk operations the execution of
+- To optimize the execution of bulk operations the execution of
   statements is delayed until the ``Sync`` message is received
 
 Copy Operations
 ---------------
 
-The COPY sub-protocol is not supported as this lacks corresponding SQL support
-in Crate.
+Crate does not support for the ``COPY`` sub-protocol.
```
Comment (matriv) [inline blackbox/docs/protocols/postgres.txt]: does not have support for the
or
does not support the

File: blackbox/docs/protocols/postgres.txt [hunk 7]
```diff
@@ -67,27 +62,25 @@ The `Extended Query`_ protocol is implemented with the following limitations:
...
-Canceling Requests in Progress
-------------------------------
+Canceling Requests
+------------------
 
-Operations can be cancelled using the ``KILL`` statement. There is no support
-for the ``CancelRequest`` message and the Server also won't send a
-``BackendKeyData`` message during connection initialization.
+Operations can be cancelled using the ``KILL`` statement, hence the
+``CancelRequest`` message  is unsupported. Consequently, the also server won't
```
Comment (matriv) [inline blackbox/docs/protocols/postgres.txt]: remove "also"

File: blackbox/docs/protocols/postgres.txt [hunk 8]
```diff
@@ -142,58 +135,54 @@ implemented::
...
 
 Connection failover and load balancing is supported as described here:
 `PostgreSQL JDBC connection failover`_.
 
 .. note::
 
-    **targetServerType** parameter shouldn't be used as for Crate there is no distinction between
-    master and slave nodes.
+    It is not recommended to use the **targetServerType** parameter since Crate
+    has no Master/Slave nodes.
```
Comment (matriv) [inline blackbox/docs/protocols/postgres.txt]: Would rephraze:
... doesn't have the concept of Master/Slave nodes
or
... in Crate there is no distinction of the nodes into master/slave

File: blackbox/docs/sql/reference/copy_from.txt [hunk 9]
```diff
@@ -120,11 +121,11 @@ Java System Properties - ``aws.accessKeyId`` and ``aws.secretKey``
 If no credentials are set the s3 client will operate in anonymous mode, see
 `AWS Java Documentation`_.
 
-Using a s3 URI will set the ``shared`` option implicitly.
+Using the 's3://' URI scheme sets the ``shared`` option implicitly.
```
Comment (matriv) [inline blackbox/docs/sql/reference/copy_from.txt]: maybe prefix and not scheme

File: blackbox/docs/sql/reference/copy_to.txt [hunk 10]
```diff
@@ -51,9 +52,8 @@ Parameters
 WHERE Clause
 ------------
 
-`WHERE` clauses follow the same syntax as used in `SELECT` statements allowing
-for partial exports that meet certain criteria. (see :ref:`sql_dql_where_clause`
-for more information).
+`WHERE` clauses use the same syntax as `SELECT` statements, allowing for
```
Comment (matriv) [inline blackbox/docs/sql/reference/copy_to.txt]: remove **for**

File: blackbox/docs/sql/reference/copy_to.txt [hunk 11]
```diff
@@ -22,11 +22,12 @@ Description
 ===========
 
 The ``COPY TO`` command exports the contents of a table to one or more
-files in the given directory, named to prevent filename conflicts. Each
-node of the cluster which contains a shard of the table will export the
-contents of its shards.
+files into a given directory with unique filenames. Each node with at least
+one shard of the table will export the contents of these onto their local disk.
```
Comment (matriv) [inline blackbox/docs/sql/reference/copy_to.txt]: you say at least one shard and then "these" which doesn't easily refer to shards

File: blackbox/docs/configuration.txt [hunk 12]
```diff
@@ -403,7 +403,7 @@ Ports
   of the node is not directly reachable from outside, e.g. running it
   behind a firewall or inside a Docker container.
 
-.. _psql.port:
+.. _psql_port:
```
Comment (matriv) [inline blackbox/docs/configuration.txt]: Are you sure this change is correct?
See failure: https://travis-ci.org/crate/crate/jobs/165487788#L1720

File: blackbox/docs/protocols/postgres.txt (hunk 1 above)
Comment (celaus) [inline blackbox/docs/protocols/postgres.txt]: nope, the label will say "Ports"

File: blackbox/docs/configuration.txt (hunk 12 above)
Comment (celaus) [inline blackbox/docs/configuration.txt]: yeah this was not about that, you can't link directly to this for whatever reason :(

File: blackbox/docs/protocols/postgres.txt (hunk 7 above)
Comment (celaus) [inline blackbox/docs/protocols/postgres.txt]: it's the "also" server 😁

File: blackbox/docs/protocols/postgres.txt [hunk 13]
```diff
@@ -142,58 +136,54 @@ implemented::
...
 
-- ``OBJECT`` and ``GEO_SHAPE`` can be streamed as ``JSON`` but require at least
-  version ``9.4.1210`` of `pgjdbc`_.
+- ``OBJECT`` and ``GEO_SHAPE`` columns can be streamed as ``JSON`` but require
+  `pgjdbc`_ version ``9.4.1210`` or newer.
 
-- Multidimensional arrays will be streamed as ``JSON`` encoded string. This is
-  done to avoid a limitation of the postgres binary streaming protocol in which
-  all sub-arrays are required to have the same length.
+- Multidimensional arrays will be streamed as ``JSON`` encoded string to avoid
```
Comment (matriv) [inline blackbox/docs/protocols/postgres.txt]: Why don't you mention the binary streaming anymore?

**Pull request merged**